libslide/src/partial_evaluator/flatten.rs

```diff
@@ -0,0 +1,139 @@
+use crate::grammar::*;
+use crate::utils::{unflatten_binary_expr, UnflattenStrategy};
+
+use std::collections::{HashMap, VecDeque};
+use std::rc::Rc;
+
+pub fn flatten_expr(expr: &Rc<Expr>) -> Rc<Expr> {
+    match expr.as_ref() {
+        Expr::Const(_) | Expr::Var(_) => Rc::clone(expr),
+        Expr::Parend(inner) | Expr::Bracketed(inner) => Rc::clone(inner),
+        Expr::BinaryExpr(BinaryExpr { op, lhs, rhs })
+            if op == &BinaryOperator::Plus || op == &BinaryOperator::Minus =>
+        {
+            flatten_add_or_sub(lhs, rhs, op == &BinaryOperator::Minus)
+        }
+        // TODO: handle everything else
+        _ => Rc::clone(expr),
+    }
+}
+
+pub fn flatten_add_or_sub(o_lhs: &Rc<Expr>, o_rhs: &Rc<Expr>, is_subtract: bool) -> Rc<Expr> {
+    let lhs = flatten_expr(o_lhs);
+    let rhs = flatten_expr(o_rhs);
+
+    let mut coeff = 0f64;
```

Comment:
this is a nitpick but we should stay consistnent.. let mut coeff: f64 = 0.0;

Reply:
good call, thanks